### PR TITLE
Fixes and updates for CubePilot PPPGW

### PR DESCRIPTION
**Pull request by bugobliterator**
* adds webui for vehicle build.
* adds PPP support to CubeOrangePlus
* changes for updated CubePilot PPPGW module
* fix broken webui trying to remove already closed socket

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/CubePilot-PPPGW/hwdef.dat [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-undef PD13
 undef PE0
 undef PE1
 
 # need to use UART8 to get RTS/CTS
 PE1 UART8_TX UART8
 PE0 UART8_RX UART8
-PD13 UART8_RTS UART8
-PD14 UART8_CTS UART8
+PA10 UART8_RTS_GPIO UART8
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/CubePilot-PPPGW/hwdef.dat:10]: RTS is always a normal GPIO (the STM32 has a bug in handling RTS in the uart peripheral, so we avoid it)

File: libraries/AP_HAL_ChibiOS/hwdef/CubeOrangePlus/hwdef.dat [hunk 2]
```diff
@@ -99,3 +99,13 @@ BOARD_VALIDATE $CHECK_IMU0_PRESENT $CHECK_IMU1_PRESENT $CHECK_IMU2_PRESENT $CHEC
...
+
+// ROMFS filesystem only
+define AP_FILESYSTEM_ROMFS_ENABLED 1
+
+// allow scripts to add parameters
+define AP_PARAM_DYNAMIC_ENABLED 1
+
+define AP_NETWORKING_BACKEND_PPP 1
+
+ROMFS_DIRECTORY libraries/AP_Networking/Web
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/CubeOrangePlus/hwdef.dat:111]: there is no libraries/AP_Networking/Web - did you forget to add?

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py [hunk 3]
```diff
@@ -186,9 +186,9 @@ def get_alt_function(self, mcu, pin, function):
                     return 0
             return None
 
-        if function and function.endswith("_RTS") and (
+        if function and (function.endswith("_RTS") or function.endswith("_RTS_GPIO") or function.endswith("_CTS_GPIO")) and (
                 function.startswith('USART') or function.startswith('UART')):
-            # we do software RTS
+            # we do software RTS/CTS
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:191]: I don't see any code changes to implement software CTS. We'd also need a much higher RTS threshold on the flight controller side (its 16 bytes at the moment, and a software CTS will be very laggy)

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py (hunk 3 above)
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py]: i've fixed it up

File: libraries/AP_HAL/utility/Socket.cpp [hunk 4]
```diff
@@ -327,8 +347,12 @@ ssize_t SOCKET_CLASS_NAME::recv(void *buf, size_t size, uint32_t timeout_ms)
     socklen_t len = sizeof(struct sockaddr_in);
     int fin = get_read_fd();
     ssize_t ret;
-    ret = CALL_PREFIX(recvfrom)(fin, buf, size, MSG_DONTWAIT, (sockaddr *)&last_in_addr[0], &len);
-    if (ret <= 0) {
+    uint32_t in_addr[4] = {};
```
Comment (peterbarker) [inline libraries/AP_HAL/utility/Socket.cpp:350]:
```suggestion
    uint32_t in_addr[4] {};
```

**Comment by tridge** [comment]
@bugobliterator I have rebased and fixed conflicts

File: libraries/AP_Networking/Web/scripts/webui.lua [hunk 5]
```diff
@@ -0,0 +1,997 @@
+--[[
```
Comment (tridge) [inline libraries/AP_Networking/Web/scripts/webui.lua:1]: why is this added? It seems to be an out of date copy of Tools/AP_Periph/Web/scripts/pppgw_webui.lua

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Networking/Web/scripts/webui.lua (hunk 5 above)
Comment (bugobliterator) [reply libraries/AP_Networking/Web/scripts/webui.lua]: removed, will create a separate method to require webui module.

**Review by tridge — APPROVED** [review]
